sw/device/tests/dif/dif_uart_test_mocked.cc

```diff
@@ -0,0 +1,734 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cassert>
+
+#include "sw/device/lib/testing/mock_mmio.h"
+
+#include "gtest/gtest.h"
+#include "sw/device/lib/base/mmio.h"
+
+extern "C" {
+#include "sw/device/lib/dif/dif_uart.h"
+#include "uart_regs.h"  // Generated.
+}  // extern "C"
+
+namespace {
+constexpr uint32_t kUint32Max = 0xffffffff;
```

Comment:
If you need the maximum of some integral type, prefer `std::numeric_limits<T>::max()`.

Reply:
Done.